keysign/util.py

```diff
@@ -29,13 +30,14 @@
 
 
 def mac_generate(key, data):
-    mac = hmac.new(key, data).hexdigest().upper()
+    mac = hmac.new(key, data, hashlib.sha256).hexdigest().upper()
     log.info("MAC of %r is %r", data[:20], mac[:20])
-    return mac
+    return mac[:20]
+
 
 def mac_verify(key, data, mac):
     computed_mac = mac_generate(key, data)
-    result = hmac.compare_digest(mac.upper(), computed_mac.upper())
```

Comment:
I'm really doubtful of this change.  My gut feeling says that it's very weird for this relatively abstract `mac_verify` function to hard code the length and, even worse, to truncate the MAC to be verified.  It feels really odd to have `mac_verify(key, data, mac)` and `mac_verify(key, data, mac+"foo")` both verify correctly.
Is it necessary to truncate here?